### PR TITLE
Updated color palette for the Uplift theme

### LINKED ISSUE #261: UI Uplift: Color Exploration
Colors need to be finalized for the March release. Danielle and Kevin have been working on colors/branding for the iOS apps, so would be good to connect with them when we finalize the ids color palette for the March release. 

### Timing Estimation 
W/o 1.7 Kayiu to partner with Danielle/Kevin on where they currently are with colors
W/o 1.14 Kayiu to partner with Danielle/Kevin for updating current color palette 
W/o 1.21 Colors should be finalized to hand off to dev by EOW
W/o 1.28 dev implements color changes into tokens 
W/o 2.4- w/o 3.15 Kayiu to put in a PR with all the changes, implement color changes into tokens 
...waiting on changing color names for future

### PR DESCRIPTION
**Pull request by kayiuho**
Updated color palette from 8 hues and 10 variations per hue to 6 hues and 5 variations per hue.

Closes #261

### CODE REVIEW

File: design-tokens/props/button.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
                         "value": "{theme.color.brand.secondary.base.value}"
                     },
                     "font": {
-                        "value": "{theme.color.palette.graphite.80.value}"
+                        "value": "{theme.color.border.daker.value}"
```
Comment (nickwynja) [inline design-tokens/props/button.json]: @clepore Typo `daker` => `darker`. There are several instances of this.

File: design-tokens/theme-uplift/theme.json [hunk 2]
```diff
@@ -3,33 +3,39 @@
...
+                    "alt":      { "value": "{theme.color.palette.azure.darker.value}" },
                     "contrast": { "value": "{theme.color.palette.white.value}" }
                 },
                 "secondary": {
-                    "base":     { "value": "{theme.color.palette.graphite.30.value}" },
-                    "alt":      { "value": "{theme.color.palette.graphite.40.value}" },
-                    "contrast": { "value": "{theme.color.palette.graphite.80.value}" }
+                    "lighter": { "value": "{theme.color.palette.slate.lighter.value}" },
+                    "base":     { "value": "{theme.color.palette.slate.base.value}" },
+                    "alt":      { "value": "{theme.color.palette.graphite.darker.value}" },
```
Comment (nickwynja) [inline design-tokens/theme-uplift/theme.json]: References `graphite` colors which no longer exist in palette. Needs to be changed to a `slate`

**Comment by nickwynja** [comment]
I'm still seeing build errors:

```
Property Reference Errors:
Reference doesn't exist: theme.color.brand.primary.base.value tries to reference theme.color.palette.azure.value, which is not defined
Reference doesn't exist: theme.color.font.muted.value tries to reference theme.color.palette.slate.value, which is not defined
Reference doesn't exist: body.color.primary.background.value tries to reference theme.color.palette.graphite.10.value, which is not defined
Reference doesn't exist: button.color.primary.disabled.background.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.primary.disabled.border.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.primary.focus.background.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.primary.focus.border.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.primary.initial.background.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.secondary.focus.background.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.secondary.focus.border.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.tertiary.disabled.font.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: button.color.tertiary.focus.font.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: button.color.tertiary.initial.font.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: button.shadow.focus.boxshadow.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: checkbox.color.checked.disabled.font.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: checkbox.color.checked.initial.background.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: checkbox.color.unchecked.disabled.font.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: checkbox.shadow.focus.boxshadow.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: dropdown.color.selected.background.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: input.color.disabled.font.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: input.color.focus.border.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: label.color.initial.font.value tries to reference theme.color.palette.graphite.50.value, which is not defined
Reference doesn't exist: link.color.disabled.font.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: link.color.initial.font.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: link.color.visited.font.value tries to reference theme.color.palette.amethyst.60.value, which is not defined
Reference doesn't exist: list.color.item.selected.background.value tries to reference theme.color.palette.graphite.10.value, which is not defined
Reference doesn't exist: radio.color.checked.disabled.background.value tries to reference theme.color.palette.graphite.10.value, which is not defined
Reference doesn't exist: radio.color.checked.initial.fill.value tries to reference theme.color.brand.primary.base.value, which is not defined
Reference doesn't exist: radio.color.unchecked.disabled.label.value tries to reference theme.color.font.muted.value, which is not defined
Reference doesn't exist: radio.shadow.focus.boxshadow.value tries to reference theme.color.brand.primary.base.value, which is not defined
```

**Comment by nickwynja** [comment]
@kayiuho Can you let me know what color tokens these should map to for `uplift`? If so, I'll work out integrating into the themes.

* `body.color.primary.background.value`
* `label.color.initial.font.value`
* `link.color.visited.font.value`
* `list.color.item.selected.background.value`
* `radio.color.checked.disabled.background.value`

**Comment by kayiuho** [comment]
* `label.color.initial.font.value` maps to `theme.color.palette.slate.darker.value`
* `link.color.visited.font.value` maps to `theme.color.palette.amethyst.darker.value`
* `list.color.item.selected.background.value` maps to `theme.color.palette.slate.base.value`
* `radio.color.checked.disabled.background.value` maps to `theme.color.palette.slate.lightest.value`

**Comment by nickwynja** [comment]
This PR contains one color change in the soho theme for consistency sake:

`label.color.initial.font.value` becomes `theme.color.font.info.value` (`theme.color.palette.graphite.60.value`) instead of `theme.color.palette.graphite.50.value`

There's one unresolved issue with `link.color.visited.font.value tries to reference theme.color.palette.amethyst.60.value, which is not defined` that is causing some existential problems that I think we need to discuss.

@clepore: this warning is also exposing an exception with the script I'd like you to take a look at:

```
(node:86598) UnhandledPromiseRejectionWarning: Error: Problems were found when trying to resolve property references
    at Object.exportPlatform (/Users/nwynja/Projects/infor-design/design-system/node_modules/style-dictionary/lib/exportPlatform.js:54:11)
    at Object.buildPlatform (/Users/nwynja/Projects/infor-design/design-system/node_modules/style-dictionary/lib/buildPlatform.js:58:21)
    at /Users/nwynja/Projects/infor-design/design-system/node_modules/style-dictionary/lib/buildAllPlatforms.js:33:10
    at /Users/nwynja/Projects/infor-design/design-system/node_modules/lodash/lodash.js:4911:15
    at Function.forIn (/Users/nwynja/Projects/infor-design/design-system/node_modules/lodash/lodash.js:12951:11)
    at Object.buildAllPlatforms (/Users/nwynja/Projects/infor-design/design-system/node_modules/style-dictionary/lib/buildAllPlatforms.js:32:5)
    at configs.forEach.config (/Users/nwynja/Projects/infor-design/design-system/scripts/node/build-tokens.js:107:12)
    at Array.forEach (<anonymous>)
    at Promise (/Users/nwynja/Projects/infor-design/design-system/scripts/node/build-tokens.js:95:13)
    at new Promise (<anonymous>)
(node:86598) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:86598) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
You should run into it when running `npm run build:tokens` on this branch. Seems like the exception needs to be caught and handled gracefully.

**Comment by nickwynja** [comment]
@kayiuho I think for backwards compatibility, we're going to have to stick with the same number of hues and colors that we had in `theme-soho`.

I'm going to close the PR but leave the branch around because there is some improvements from this that we can migrate.

Can you please revisit this using the same base color and 10 overall variations for each?

**Comment by kayiuho** [comment]
Will do. I'll get that going with adding 5 more variations per color. Can we do 6 hues? or we have to go back to 8 hues too?

**Comment by nickwynja** [comment]
I think we need to stick with 8. Let's start with a 1:1 match of the list of colors we had in Soho and then can reduce from there if possible.

**Pull request closed without merging**